store/builder.go

```diff
@@ -0,0 +1,381 @@
+func (d *Builder) buildEigenDAV2Backend(
+	ctx context.Context,
+	maxBlobSizeBytes uint,
+	eigenDACertVerifierAddress string,
+) (common.GeneratedKeyStore, error) {
+	// TODO: Figure out how to better manage the v1 verifier
+	//  may make sense to live in some global kzg config that's passed down across EigenDA versions
+	kzgProver, err := prover.NewProver(d.v1VerifierCfg.KzgConfig, nil)
+	if err != nil {
+		return nil, fmt.Errorf("new KZG prover: %w", err)
+	}
+
+	if d.memConfig != nil {
+		return memstore_v2.New(d.ctx, d.log, d.memConfig, kzgProver.Srs.G1)
+	}
+
+	ethClient, err := d.buildEthClient()
+	if err != nil {
+		return nil, fmt.Errorf("build eth client: %w", err)
+	}
+
+	relayPayloadRetriever, err := d.buildRelayPayloadRetriever(ethClient, maxBlobSizeBytes, kzgProver.Srs.G1)
+	if err != nil {
+		return nil, fmt.Errorf("build relay payload retriever: %w", err)
+	}
+
+	certVerifier, err := verification.NewCertVerifier(
+		d.log, ethClient, d.v2ClientCfg.BlockNumberPollIntervalDuration)
+	if err != nil {
+		return nil, fmt.Errorf("new cert verifier: %w", err)
+	}
+
+	payloadDisperser, err := d.buildPayloadDisperser(ctx, ethClient, kzgProver, certVerifier)
+	if err != nil {
+		return nil, fmt.Errorf("build payload disperser: %w", err)
+	}
+
+	v2Config := &eigenda_v2.Config{
+		CertVerifierAddress: eigenDACertVerifierAddress,
+		MaxBlobSizeBytes:    uint64(maxBlobSizeBytes),
+		PutRetries:          d.v2ClientCfg.PutRetries,
+	}
+
+	return eigenda_v2.NewStore(d.log, v2Config, payloadDisperser, relayPayloadRetriever, certVerifier)
+}
+
+// buildEigenDAV1Backend ... Builds EigenDA V1 storage backend
+func (d *Builder) buildEigenDAV1Backend(
+	ctx context.Context,
+	putRetries uint,
+	maxBlobSize uint) (common.GeneratedKeyStore, error) {
+	verifier, err := verify.NewVerifier(&d.v1VerifierCfg, d.log)
+	if err != nil {
+		return nil, fmt.Errorf("new verifier: %w", err)
+	}
+
+	if d.v1VerifierCfg.VerifyCerts {
+		d.log.Info("Certificate verification with Ethereum enabled")
+	} else {
+		d.log.Warn("Certificate verification disabled. This can result in invalid EigenDA certificates being accredited.")
+	}
+
+	if d.memConfig != nil {
+		d.log.Info("Using memstore backend for EigenDA V1")
+		return memstore.New(ctx, verifier, d.log, d.memConfig)
+	}
+	// EigenDAV1 backend dependency injection
+	var client *clients.EigenDAClient
+	d.log.Warn("Using EigenDA backend.. This backend type will be deprecated soon. Please migrate to V2.")
+	client, err = clients.NewEigenDAClient(d.log, d.v1EdaClientCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return eigenda.NewStore(
+		client,
+		verifier,
+		d.log,
+		&eigenda.StoreConfig{
+			MaxBlobSizeBytes:     uint64(maxBlobSize),
+			EthConfirmationDepth: d.v1VerifierCfg.EthConfirmationDepth,
+			StatusQueryTimeout:   d.v1EdaClientCfg.StatusQueryTimeout,
+			PutRetries:           putRetries,
+		},
+	)
+}
+
+// BuildManager ... Builds storage manager object
+func (d *Builder) BuildManager(
```

Comment:
Move Public methods to top of file, right below constructor. Otherwise I have to search for this function, which is really the only important one

Reply:
I did this when I cleaned up the builder in [bcd2379](https://github.com/Layr-Labs/eigenda-proxy/pull/321/commits/bcd237941f223348dab947e3aa6ca6820d300232)